pkg/controller/storagecluster/storagecluster.go

```diff
@@ -649,6 +658,45 @@ func (c *Controller) deleteStorageCluster(
 	return nil
 }
 
+func (c *Controller) removeResources(namespace string) error {
+	objs, err := k8s.GetAllObjects(c.client, namespace)
+	if err != nil {
+		return err
+	}
+
+	for _, obj := range objs {
+		gcAnnotated := false
+		if obj.GetAnnotations() != nil {
+			gcAnnotated, _ = strconv.ParseBool(obj.GetAnnotations()[constants.AnnotationGarbageCollection])
+		}
+
+		if gcAnnotated || c.gcNeeded(obj) {
+			logrus.Infof("Garbage collect object %s %s", obj.GetObjectKind().GroupVersionKind().Kind, obj.GetName())
+			if err := c.client.Delete(context.TODO(), obj); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// All external object should use the GC annotation for operator to delete them.
+// For backward compatibility, we still delete some objects without the annotation.
+func (c *Controller) gcNeeded(obj client.Object) bool {
+	if obj == nil {
+		return false
+	}
+
+	if obj.GetObjectKind().GroupVersionKind().Kind == "ConfigMap" {
+		if obj.GetName() == "px-attach-driveset-lock" ||
+			strings.HasPrefix(obj.GetName(), "px-bringup-queue-lock") {
+			return true
+		}
+	}
+
+	return false
+}
+
```

Comment:
This is very portworx specific. Can we move at least this or the entire removeResources function to the portworx driver? Let's not spill px specific code in the controller.